### PR TITLE
Cutout Tool and Flag+Comment Boxes

### PR DESCRIPTION
**Pull request by robelgeda**
New:
- MOSViz comments and flags.
- Cutout Tool for JWST/NIRSpec MSA Programs.
- MOSViz Table Generator for JWST/NIRSpec MSA Programs.

Changes:
- Handle selections with no cutout images.
        - `image_widget.setVisible(False)` when image base-name is "None".

Bug Fixes:
- MOSVizViewer._update_data_components:
        - Order of operations causes an inf loop.
        - Messages sent in hub cause a loop.
        - Fix: [`self._loaded_data[key] = data `](https://github.com/robelgeda/mosviz/blob/2bcb75161cee4e1875d4937f8d097af41469337d/mosviz/viewers/mos_viewer.py#L507)(Line 507) before appending data to dc.

### CODE REVIEW

File: mosviz/plugins/cutout_tool.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+from astropy.io import fits
+from astropy.wcs import WCS, NoConvergence
+from astropy.coordinates import SkyCoord
+from astropy.nddata.utils import (Cutout2D, NoOverlapError)
+from astropy import log
+from astropy.coordinates import Angle 
+
+from .. import UI_DIR       
+
+class cutoutTool (QMainWindow):
```
Comment (robelgeda) [inline mosviz/plugins/cutout_tool.py]: Will fix this ASAP. Thanks for the heads up.

File: mosviz/plugins/cutout_tool.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+            self.savePath = self.inSpectra.text() 
+            if self.savePath == "":
+                self.inSave.setText("")
+            else:
+                self.inSave.setText(os.path.join(self.savePath,"[Name]_cutouts",""))
+
+    def custom_path(self):
+        info = "Changing the default save path will result in absolute paths to be saved into the MOSViz Table."
+        info+= "This means the MOSViz Table will only work on your computer and cannot be shared."
+        info = QMessageBox.information(self, "Status:", info)
```
Comment (robelgeda) [inline mosviz/plugins/cutout_tool.py]: That is awesome! Will make the edit.

**Comment by robelgeda** [comment]
All changes have been applied for this PR.

File: mosviz/plugins/cutout_tool.py [hunk 3]
```diff
@@ -136,11 +136,17 @@ def collect_text(self):
             self.save_path  = self.spec_path  
 
         if self.xSize.text() != "":
-            self.cutout_x_size = float(self.xSize.text())
+            try:
+                self.cutout_x_size = float(self.xSize.text())
+            except:
```
Comment (nmearl) [inline mosviz/plugins/cutout_tool.py]: You should make the exception explicit (I'm assuming in this case you're expecting that the string might be a float, and will fail with a `ValueError` exception) or use a `QValidator`.

**Comment by nmearl** [comment]
Failing build due to issue with specutils and possibly astropy helpers. Ignoring.

**Pull request merged**